src/iptvsimple/utilities/StreamUtils.cpp

```diff
@@ -63,7 +63,41 @@ void StreamUtils::SetAllStreamProperties(PVR_NAMED_VALUE* properties, unsigned i
     }
     else // inputstream.adpative
     {
-      StreamUtils::SetStreamProperty(properties, propertiesCount, propertiesMax, PVR_STREAM_PROPERTY_STREAMURL, streamURL);
+      // If no media headers are explicitly set for inputstream.adaptive,
+      // strip the headers from streamURL and put it to media headers property
+
+      if (channel.GetProperty("inputstream.adaptive.stream_headers").empty()) {
```

Comment:
Brackets on the following line please. The same as the rest of the code.

Reply:
> Can you give some examples of URLs and what the value of `stream_headers` ends up being?

Let me start the test VM up and send you the debug log by email.

Reply:
done